api/src/test/java/org/openmrs/module/cohort/api/dao/CohortGenericDaoTest.java

```diff
@@ -147,4 +149,12 @@ public void getByUuid_shouldReturnNullForVoidedOrRetiredCohort() {
 		
 		assertThat(dao.get(COHORT_UUID), nullValue());
 	}
+	
+	@Test
+	public void shouldVoidCohortM() {
+		CohortM cohortToVoid = dao.get(COHORT_UUID);
+		Context.getService(CohortService.class).voidCohortM(cohortToVoid, "delete cohort");
+		CohortM voidedCohort = dao.createOrUpdate(cohortToVoid);
```

Comment:
Why are you calling `dao.createOrUpdate()` after `service. voidCohortM()`? Isn't the service method the one supposed to take care of the voiding?

Reply:
I just call `dao.createOrUpdate` because it has used in `voidCohortM` method.